crypto/bn/bn_gcd.c

```diff
@@ -611,18 +612,30 @@ int BN_gcd(BIGNUM *r, const BIGNUM *in_a, const BIGNUM *in_b, BN_CTX *ctx)
         goto err;
 
     /* find shared powers of two, i.e. "shifts" >= 1 */
+    pow2_flag = 1;
+    pow2_chunk_index = 0;
+    pow2_shifts = 0;
     for (i = 0; i < r->dmax && i < g->dmax; i++) {
-        mask = ~(r->d[i] | g->d[i]);
-        for (j = 0; j < BN_BITS2; j++) {
-            bit &= mask;
-            shifts += bit;
-            mask >>= 1;
-        }
+        pow2_mask = r->d[i] | g->d[i];
+        pow2_temp = !constant_time_is_zero(pow2_flag) & !constant_time_is_zero_64(pow2_mask);;
+        pow2_flag &= !pow2_temp;
+        pow2_temp = ((~pow2_temp & (pow2_temp - 1)) >> (BN_BITS2 - 1)) - 1; // https://github.com/openssl/openssl/blob/067fbc01b9e867b31c71091d62f0f9012dc9e41a/crypto/bn/bn_lib.c#L950C5-L950C74
+        pow2_shifts += 1 & pow2_temp;
+        pow2_temp = (i ^ pow2_chunk_index) & pow2_temp;
+        pow2_chunk_index ^= pow2_temp;
+    }
+    pow2_shifts *= BN_BITS2;
+    pow2_mask = r->d[pow2_chunk_index] | g->d[pow2_chunk_index];
```

Comment:
This dereference won't be constant time either because here you're indexing based on the values of the numbers.

You need to calculate the pow2_mask value within the for loop above.